ssl/bio_ssl.c

```diff
@@ -117,10 +117,10 @@ static int ssl_read(BIO *b, char *buf, size_t size, size_t *readbytes)
             }
         }
         if ((sb->renegotiate_timeout > 0) && (!r)) {
-            unsigned long tm;
+            time_t tm;
 
-            tm = (unsigned long)time(NULL);
-            if (tm > sb->last_time + sb->renegotiate_timeout) {
+            tm = time(NULL);
+            if (tm - sb->last_time > (time_t)sb->renegotiate_timeout) {
```

Comment:
my approach had the nice added value that it is robust if the system time is set back,
that's why I rewrote the condition to `tm - bs->last_time > bs->renegotiate_timeout`
I thought that's obvious.